### PR TITLE
For #7955 - Changes fonts according to abstract.

### PR DESCRIPTION
**Pull request by codrut-topliceanu**
### Pull Request checklist

- [x] **Tests**: This PR doesn't include tests, not needed for font change.
- [x] **Screenshots**: This PR includes a screenshot.
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md)

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/res/layout/activity_migration.xml [hunk 1]
```diff
@@ -33,7 +34,7 @@
                 android:layout_marginStart="24dp"
                 android:maxLines="2"
                 android:text="@string/migration_title"
-                android:fontFamily="sans-serif-medium"
+                android:fontFamily="Sharp Sans"
```
Comment (pocmo) [inline app/src/main/res/layout/activity_migration.xml]: I have never seen this font name. Is that a font we ship with Fenix? @ekager @sblatz maybe you can review this change?

File: app/src/main/res/layout/activity_migration.xml [hunk 2]
```diff
@@ -46,16 +47,17 @@
...
                 android:text="@string/migration_description"
                 android:textColor="@color/text_scale_example_text_color"
                 android:textSize="16sp"
-                android:textStyle="bold"
-                android:fontFamily="sans-serif-light"
                 app:layout_constraintEnd_toEndOf="parent"
                 app:layout_constraintStart_toStartOf="parent"
-                app:layout_constraintTop_toBottomOf="@+id/migration_firefox_logo" />
+                app:layout_constraintTop_toBottomOf="@+id/migration_firefox_logo"
+                tools:targetApi="o" />
```
Comment (pocmo) [inline app/src/main/res/layout/activity_migration.xml]: What does `tools:targetApi="o"` mean here?

File: app/src/main/res/layout/migration_list_item.xml [hunk 3]
```diff
@@ -27,7 +27,7 @@
             android:layout_marginStart="24dp"
             android:layout_marginEnd="32dp"
             android:textColor="@color/dark_grey_90"
-            android:fontFamily="sans-serif-medium"
+            android:fontFamily="Inter UI"
```
Comment (pocmo) [inline app/src/main/res/layout/migration_list_item.xml]: Same here. CC @ekager @sblatz

File: app/src/main/res/layout/activity_migration.xml (hunk 2 above)
Comment (codrut-topliceanu) [reply app/src/main/res/layout/activity_migration.xml]: It's to suppress the warning regarding `android:justificationMode="inter_word"` which needs API level 26 or higher.

File: app/src/main/res/layout/activity_migration.xml (hunk 1 above)
Comment (codrut-topliceanu) [reply app/src/main/res/layout/activity_migration.xml]: I've found this font in use in `fragment_crash_reporter.xml`. That's why I took the liberty to use it again.

File: app/src/main/res/layout/migration_list_item.xml (hunk 3 above)
Comment (codrut-topliceanu) [reply app/src/main/res/layout/migration_list_item.xml]: This one I saw used in `styles.xml` under `<style name="SearchShortcutsLabelStyle">`

File: app/src/main/res/layout/activity_migration.xml (hunk 1 above)
Comment (sblatz) [reply app/src/main/res/layout/activity_migration.xml]: @AmyYLee what font should we be using here?

**Pull request closed without merging**